vms/evm/uptimetracker/state.go

```diff
@@ -138,33 +132,20 @@ func (s *state) GetStartTime(nodeID ids.NodeID) (time.Time, error) {
 
 func (s *state) addValidatorUpdate(vdr *validator) {
 	s.addValidator(vdr.validationID, vdr)
-	s.updates[vdr.validationID] = updatedStatus
+	s.updatedValidators.Add(vdr.validationID)
 }
 
-func (s *state) updateValidator(
-	validationID ids.ID,
-	isActive bool,
-	weight uint64,
-) error {
-	v, ok := s.validators[validationID]
-	if !ok {
-		return database.ErrNotFound
-	}
+func (s *state) updateValidator(validationID ids.ID, isActive bool) bool {
+	v := s.validators[validationID]
 
-	updated := deletedStatus
```

Comment:
We never handle deletes here - in the calling code we check that we have this validator.